Учебный проект: покоритель файлов

akrjuger opened the pull request:
---
:mortar_board: [Учебный проект: покоритель файлов](https://up.htmlacademy.ru/nodejs/3/user/521509/tasks/8)

Commit:
fix typo in constant (thanks Nikita)

Commit:
change arrays to files

Baristanko commented on src/service/cli/generate.js:
```diff
@@ -103,6 +99,10 @@ const SumRestrict = {
   MAX: 100000,
 };
 
+let categories;
```
нужно избавиться от этих глобальный переменных.
Их нужно передавать в generateOffer через параметры.

Baristanko commented on src/service/cli/generate.js:
```diff
@@ -154,6 +154,24 @@ const generateOffers = (count) => {
+    return contentArray;
+  } catch (error) {
+    await Promise.reject(filePath);
+    return [];
```
так не будет работать, лучше сделать так:
в catch сделать
```js
throw new Error(`Не удалось прочитать файл ${filePaath}: ${error.message}`)
```
делать await Promise.reject() не нужно,
 возвращать после него какие-либо значения тоже, код который вызвал этот промис никогда их не получит.
 В случае ошибки в промисе он автоматически делает reject с объектом ошибки

Baristanko commented on src/service/cli/generate.js:
```diff
@@ -173,8 +191,28 @@ module.exports = {
+    // Комментарий-вопрос наставнику: я нашел, что можно так делать, что результат промисов возвращается в том же порядке, как они отдавались в промис алл.
+    // Тесты мои показали, что все работает так как нужно.
+    // Но меня немного смущает это конструкция ниже. Может можно изящнее?
+    try {
```
когда мы избавимся от глобальный переменных можно сделать так:
```js
const [ categories, offerTitles, saleTitles, sentences ] = await Promise.all(...);
```
Вообще в этой функции достаточно одного  try/catch

Baristanko reviewed (commented):
Лучше вопросы задавать не в комментариях кода, а писать в телегу

Commit:
fixes after mentors review:
1) delete global variables
2) delete unnecessary try, catch
3) replace Promise.reject --> throw new Error

Baristanko reviewed (approved):
Да теперь все хорошо. У JSON.stringify есть третий аргумент отвечачающий за prettyPrint:
`JSON.stringify(data, null, 4)`
Выведет  json  в более красивом формате. Это я для общего развития.
